### PR TITLE
feat: add filter params

### LINKED ISSUE #167: Feature: Filtering should be reflected in the URL
### Type of feature

🍕 Feature

### Current behavior

When selecting a filter it requests new data from the API (supabase for now). It should provide the filter in the URL. i.e. `hot.opensauced.pizza?filter=upvoted`. The default can still be popular, but it should also respect the URL 👉🏾`hot.opensauced.pizza?filter=popular`

### Suggested solution

Use next.js to add query params. 

### Additional context

https://nextjs.org/docs/routing/dynamic-routes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by htuerker**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

1. Added react-router dependency
2. Wrapped `App` component by `BrowserRouter` to use router functionalities into further children. 
3. Created `searchParams` state by `useSearchParams`
4. Used `activeLink` as a derived state from `searchParams` instead of a managed state.
5. Moved the handler logic into where we handle state
6. Removed unused props from `SecondaryNav`

Fixes #167 

## Mobile & Desktop Screenshots/Recordings

Screen Shot 2022-06-17 at 18 24 00

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by 0-vortex** [comment]
The pull request code looks great however I was hoping we can use actual routes, not query parameters - at least that was my intention in commenting here: https://github.com/open-sauced/hot/issues/167#issuecomment-1132346093

Would you be able and willing to transform the query parameters into something like:
`localhost:3000/?filter=upvoted` into `localhost:3000/most-upvoted-repositories`?

**Comment by htuerker** [comment]
> Would you be able and willing to transform the query parameters into something like: `localhost:3000/?filter=upvoted` into `localhost:3000/most-upvoted-repositories`?

This would be a little tricky but if I got correct you want to see a meaningful pathname instead of query params. So, your suggestion is to not use query parameters at all.

Instead of the button with a handler in `SecondaryNav`, we can use `<Link to="/most-upvoted-repositories />` to navigate, then parse the current pathname with a simple hash. `parsePathName("most-upvoted-repositories") ===> return "upvoted"`.

**Comment by htuerker** [comment]
I delivered changes by last commit @0-vortex 

Last changes in order:

1. Replace filtering buttons with `Links`
2. `activeLink` is a derived state linked to location.
3. `activeLink` state already protected by unknown paths i.e `/some-user-given-path` defaults to `popular`
4. `limit` is a derived state linked to `searchParams`
5. `limit` state defaults `0` and `NaN` values to `25`
6. Both are generated synchronously so there's no state batching problem as I know
7. When the location changes via `SecondaryNav` links `limit` defaults to `25` (this is the reason why I changed `limit` logic)

Problems:
1. You can give any number you want as a limit in the path. But this is a pagination problem so the new API should correct this problem. We can easily update this implementation with `per_page` `page_number` logic.

**Comment by bdougie** [comment]
@htuerker I think what you have works great. It is a good first step. My only suggestion is there are too many word. `most-popular-repositories` should be `popular`.

File: src/components/PostsWrap.tsx [hunk 1]
```diff
@@ -6,19 +6,46 @@ import GridDisplay from './GridDisplay';
...
 import useSupabaseAuth from '../hooks/useSupabaseAuth';
+import { useLocation, useSearchParams } from 'react-router-dom';
 
 interface PostWrapProps{
   textToSearch: string
 }
 
-const PostsWrap = ({ textToSearch } :PostWrapProps): JSX.Element => {
+const locationsHash: { [index: string] : string | undefined } = {
+  "/most-popular-repositories": "popular",
```
Comment (bdougie) [inline src/components/PostsWrap.tsx:16]: Please use the filter name instead. 

```suggestion
  "/popular": "popular",
```

**Review by bdougie — CHANGES_REQUESTED** [review]

**Review by bdougie — APPROVED** [review]
Sorry was out of town and this fell the list of things to review.

**Pull request merged**